tests/specs/mcd/verification.k

```diff
@@ -99,4 +99,23 @@ module VERIFICATION
     rule M:Memory [ K <- V ] [ K' <- V' ]             => M [ K' <- V' ] [ K <- V ]       requires K' <Int K
     rule M:Memory [ K := BUF:ByteArray ] [ K' <- V' ] => M [ K' <- V' ] [ K := BUF ]     requires K' <Int K orBool K +Int #sizeByteArray(BUF) <=Int K'
 
+    // ### dstoken-transferfrom-fail-rough
+
+    rule hash2(_, hash2(_, _)) ==K hash2(_, N) => false requires N <=Int 10
+
+    rule chop(N +Int M) <Int N => true requires #rangeUInt(256, N) andBool #rangeUInt(256, M) andBool notBool #rangeUInt(256, N +Int M)
+
+    // ### dsvalue-read-pass
+
+    rule M [ K <- V ] => M requires #lookup(M, K) ==Int V
+
+    // ### cat-ilks-pass
+
+    rule chop(hash2(I, I') +Int N) => hash2(I, I') +Int N requires N <Int 3
+
+    rule hash2(_, _) +Int N ==K hash2(_, _) +Int M => false requires N =/=Int M
```

Comment:
This is better to be expressed as equality decomposition, like:
```
rule hash2(A, B) +Int N ==K hash2(C, D) +Int M => hash2(A, B) ==K hash2(C, D) andBool N ==K M
  requires #range(0 <=N <10) andBool #range(0 <=M <10)
```
Ranges are required, otherwise lemma is unsound.